10-std.tex

```diff
@@ -295,7 +295,79 @@ \subsection{Extensions}
 \section{Lambda}
 \label{std-Lambda}
 
+\define{Lambda}{define-lambda}{Lambda is a functional language concept within Haxe that allows you to apply a function to a set or \tref{custom iterators}{lf-iterators}. The Lambda class is a collection of functional methods in order to use functional-style programming with Haxe.}
 
+It is ideally used with \expr{using Lambda} (read \tref{Static Extension}{lf-static-extension}) and then acts as an extension to \type{Iterable} types. 
+
+On static platforms, working with the \type{Iterable} structure might be slower than performing the operations directly on known types, such as \type{Array} and \type{List}.
+
+\paragraph{Lambda Functions}
+The Lambda class allows us to operate on an entire \type{Iterable} at once.
+This is often preferable to looping routines since it is less error prone and easier to read. 
+For convenience, the \type{Array} and \type{List} class contains some of the frequently used methods from the Lambda class.
+
+It is helpful to look at an example. The exists function is specified as:
+
+\begin{lstlisting}
+static function exists<A>( it : Iterable<A>, f : A -> Bool ) : Bool
+\end{lstlisting}
+
+Most Lambda functions are called in similar ways. The first argument for all of the Lambda functions is the \type{Iterable} on which to operate. Many also take a function as an argument.
+
+\begin{description}
+	\item[\expr{Lambda.array}, \expr{Lambda.list}] Convert Iterable to \type{Array} or \type{List}. It returns a copy if it's already of that type.
+	\item[\expr{Lambda.count}] Count the number of elements.  If the Iterable is a \type{Array} or \type{List} it is faster to use its length property.
+	\item[\expr{Lambda.empty}] Determine if the Iterable is empty. For all Iterables it is best to use the this function; it's also faster than compare the length (or result of Lambda.count) to zero.
```

Comment:
It comes from the old documentation. I haven't tested it. I _think_ it should be faster than Lambda.count since that first iterates over the full list.